pyproject.toml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[project]
+name = "github-runner-image-builder"
+version = "0.0.1"
+authors = [{ name = "Yanks Yoon", email = "yangsoo.yoon@canonical.com" }]
+description = "A github runner image builder package"
+readme = "README.md"
+requires-python = ">=3.10"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+]
+dynamic = ["dependencies"]
+[tool.setuptools.dynamic]
+dependencies = { file = ["requirements.txt"] }
```

Comment:
@javierdelapuente Netbox can try this?

Reply:
Thanks for the idea!. I was reviewing https://github.com/canonical/craft-parts/blob/main/craft_parts/plugins/python_plugin.py#L115, and if there is a `pyproject.toml`, the rockcraft python plugin always runs `pip install .`, which tries to build the wheel (the problem in NetBox is building the wheel, not that the dependencies are in requirements.txt). Interestenly, I ended up in a very related discussion in here: https://discuss.python.org/t/projects-that-arent-meant-to-generate-a-wheel-and-pyproject-toml/29684/193 and https://python-poetry.org/docs/pyproject/